swift/internal/developer_frameworks.bzl

```
@@ -0,0 +1,119 @@
+""" Functions for fetching developer framework paths """
+
+load("@bazel_skylib//lib:paths.bzl", "paths")
+load(
+    ":utils.bzl",
+    "compact",
+    "is_xcode_at_least_version",
+)
+
+def platform_developer_framework_dir(
+        apple_toolchain,
+        apple_fragment,
+        xcode_config):
+    """Returns the Developer framework directory for the platform.
+
+    Args:
+        apple_fragment: The `apple` configuration fragment.
+        apple_toolchain: The `apple_common.apple_toolchain()` object.
+        xcode_config: The Xcode configuration.
+
+    Returns:
+        The path to the Developer framework directory for the platform if one
+        exists, otherwise `None`.
+    """
+
+    # All platforms have a `Developer/Library/Frameworks` directory in their
+    # platform root, except for watchOS prior to Xcode 12.5.
+    platform_type = apple_fragment.single_arch_platform.platform_type
+    if (
+        platform_type == apple_common.platform_type.watchos and
+        not is_xcode_at_least_version(xcode_config, "12.5")
+    ):
+        return None
+
+    return apple_toolchain.platform_developer_framework_dir(apple_fragment)
+
+def swift_developer_lib_dir(
+        apple_toolchain,
+        apple_fragment,
+        xcode_config):
+    """Returns the directory containing extra Swift developer libraries.
+
+    Args:
+        apple_fragment: The `apple` configuration fragment.
+        apple_toolchain: The `apple_common.apple_toolchain()` object.
+        xcode_config: The Xcode configuration.
+
+    Returns:
+        The directory containing extra Swift-specific development libraries and
+        swiftmodules.
+    """
+
+    platform_framework_dir = platform_developer_framework_dir(
+        apple_toolchain,
+        apple_fragment,
+        xcode_config,
+    )
+
+    return paths.join(
+        paths.dirname(paths.dirname(platform_framework_dir)),
+        "usr",
+        "lib",
+    )
+
+def developer_framework_paths(
+        apple_fragment,
+        xcode_config):
+    """Returns the developer framework paths for the given apple fragment and xcode configuration.
```

Please wrap this, and all files to 80 characters. In this case, since it needs to be one line, you can do it like this:
```suggestion
    """Returns the developer framework paths for the given apple fragment and \
    xcode configuration.
```